dashboard/app/helpers/users_helper.rb

```diff
@@ -132,7 +132,11 @@ def get_pages_completed(user, sl)
     if level.is_a? LevelGroup
       pages_completed = []
 
-      last_attempt = JSON.parse(user.last_attempt(level).level_source.data)
+      if user.last_attempt(level).level_source
```

Comment:
I'd suggest removing the condition. Then on line 158, begin the conditional with last_attempt